applications/app/controllers/SignupPageController.scala

```diff
@@ -100,29 +87,35 @@ class SignupPageController(
         val dataModel =
           DotcomNewslettersPageRenderingDataModel.apply(page, newsletters, request)
         val dataJson = DotcomNewslettersPageRenderingDataModel.toJson(dataModel)
-        common.renderJson(dataJson, page).as("application/json")
+        Future.successful(common.renderJson(dataJson, page).as("application/json"))
       }
       case Left(e) =>
         log.error(s"API call to get newsletters failed: $e")
         throw new RuntimeException()
     }
   }
 
-  private def renderNewslettersJson()(implicit
+  private def notFoundPage()(implicit
       request: RequestHeader,
-  ): Result = {
-    Cached(CacheTime.NotFound)(Cached.WithoutRevalidationResult(NotFound))
+  ): Future[Result] = {
+    Future(Cached(CacheTime.NotFound)(Cached.WithoutRevalidationResult(NotFound)))
   }
 
-  def renderNewslettersJson()(implicit
+  def renderNewsletters()(implicit
       executionContext: ExecutionContext = this.executionContext,
   ): Action[AnyContent] =
     csrfAddToken {
-      Action { implicit request =>
-        if (request.forceDCR || UseDcrNewslettersPage.isSwitchedOn) {
-          renderDCRNewslettersJson()
-        } else {
-          renderNewslettersJson()
+      Action.async { implicit request =>
+        val useDCR = request.forceDCR || UseDcrNewslettersPage.isSwitchedOn
```

Comment:
Not a blocker as this is the current behaviour, but with this logic, it’s impossible to force DCR off when the `UseDcrNewslettersPage` switch is on.

Reply:
🤷